openedx_learning/apps/authoring/publishing/api.py

```diff
@@ -216,10 +216,21 @@ def get_all_drafts(learning_package_id: int, /) -> QuerySet[Draft]:
     )
 
 
-def get_entities_with_unpublished_changes(learning_package_id: int, /) -> QuerySet[PublishableEntity]:
+def get_entities_with_unpublished_changes(
+    learning_package_id: int,
+    /,
+    include_deleted_drafts: bool = False
+) -> QuerySet[PublishableEntity]:
+    """
+    Fetch entities that have unpublished changes.
+
+    By default, this excludes soft-deleted drafts but can be included using include_deleted_drafts option.
+    """
     return PublishableEntity.objects \
-                            .filter(learning_package_id=learning_package_id) \
-                            .exclude(draft__version=F('published__version'))
+                            .filter(
+                                learning_package_id=learning_package_id,
+                                draft__version__isnull=include_deleted_drafts,
```

Comment:
Oh wait, if `include_deleted_drafts=True` in this code, it's going to _only_ return soft deletes, instead of just including soft-deletes.

Reply:
@ormsbee Ahh yes! my bad.

Reply:
Fixed.